tools/cli/src/mentat_cli/repl.rs

```diff
@@ -198,10 +207,51 @@ impl Repl {
         }
     }
 
-    fn read_file(&self, files: Vec<String>) {
+    fn read_files(&mut self, files: Vec<String>) {
         for file in files {
-            println!("Executing edn in file {}", file);
+            let res = self.read_file(file);
+            if res.is_err() {
+                match res.unwrap_err() {
+                    cli::Error(err, _) => { println!("{}", err) },
+                }
+            }
+        }
+    }
+
+    fn read_file(&mut self, file: String) -> Result<(), cli::Error> {
+        let path = Path::new(&file);
+        let display = path.display();
+
+        let f = match File::open(&path) {
+            Err(err) => bail!(cli::ErrorKind::FileError(display.to_string(), err.to_string())),
```

Comment:
Don't turn everything to strings.  It's hard to pattern match strings; it's possible (although not always easy) to pattern match structured data.  I understand that this can be difficult with `error-chain`, but if that's so, ask for help!